test/unittests/evm_test.cpp

```diff
@@ -777,35 +777,174 @@ TEST_F(evm, selfdestruct)
 
 TEST_F(evm, selfdestruct_with_balance)
 {
-    auto code = "6000ff";
-    msg.destination.bytes[0] = 1;
+    const auto beneficiary = evmc::address{{0xbe}};
+    const auto code = push({beneficiary.bytes, sizeof(beneficiary)}) + OP_SELFDESTRUCT;
+    msg.destination = evmc_address{{0x5e}};
+
+
+    accounts[msg.destination].set_balance(0);
+
+    rev = EVMC_HOMESTEAD;
+    execute(3, code);
+    EXPECT_GAS_USED(EVMC_SUCCESS, 3);
+    EXPECT_EQ(result.gas_left, 0);
+    ASSERT_EQ(recorded_account_accesses.size(), 1);
+    EXPECT_EQ(recorded_account_accesses[0], msg.destination);  // Selfdestruct.
+    recorded_account_accesses.clear();
+
+    rev = EVMC_TANGERINE_WHISTLE;
```

Comment:
Should we also validate gas left in this test?

Reply:
I don't think this is required. In many other tests it is verified that the relation between gas_used and gas_left is correct.